### PR TITLE
Update: TypeScript 2.6.2, react typings

### PR DESCRIPTION
**Pull request by dzearing**
Found real issues in the (lack of) typings which caused some breaks when updating things to the new TypeScript.

Also updating the React typings to latest 15.x.

### CODE REVIEW

File: packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx [hunk 1]
```diff
@@ -101,14 +101,6 @@ export class BaseSelectedItemsList<T, P extends IBaseSelectedItemsListProps<T>>
     let newItems = newProps.selectedItems;
 
     if (newItems) {
-      let focusIndex: number;
-
-      // If there are less new items than old items then something was removed and we
-      // should try to keep focus consistent
-      if (newItems.length < this.state.items.length) {
```
Comment (christiango) [inline packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx]: hmm, what's the background on this change?

Comment (christiango) [reply packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx]: Oh unused locals FTW!

File: packages/experiments/tsconfig.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "sourceMap": true,
     "experimentalDecorators": true,
     "importHelpers": true,
+    "noUnusedLocals": true,
```
Comment (christiango) [inline packages/experiments/tsconfig.json]: We should create a base tsconfig and have all projects extend that base config

File: packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx (hunk 1 above)
Comment (dzearing) [reply packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx]: Yes!!

Comment (dzearing) [reply packages/experiments/src/components/SelectedItemsList/BaseSelectedItemsList.tsx]: Yes!!

File: packages/experiments/tsconfig.json (hunk 2 above)
Comment (dzearing) [reply packages/experiments/tsconfig.json]: We have that...

Comment (dzearing) [reply packages/experiments/tsconfig.json]: Oops nevermind thats tslint

**Pull request merged**